socs/agents/camera_rtsp/agent.py

```diff
@@ -0,0 +1,887 @@
+            sq_x = self.width - self.sq_half
+        if sq_x < self.sq_half:
+            sq_x = self.sq_half
+        self.sq_x = sq_x
+        self.sq_y = sq_y
+
+    def grab(self):
+        img = np.zeros((self.height, self.width, 3), dtype=np.uint8)
+        img[:, :, 0] = 127
+        img[:, :, 1] = 127
+        img[:, :, 2] = 127
+        if self.frame_count % self.fixed_frames == 0:
+            self._random_sq_pos()
+        img[
+            self.sq_y - self.sq_half: self.sq_y + self.sq_half,
+            self.sq_x - self.sq_half: self.sq_x + self.sq_half,
+            0,
+        ] = 255
+        self.current = img
+        self.frame_count += 1
+        return True
+
+    def retrieve(self):
+        img = np.array(self.current)
+        self.current = None
+        return True, img
+
+    def read(self):
+        _ = self.grab()
+        return self.retrieve()
+
+    def release(self):
+        pass
+
+
+class CameraRTSPAgent:
+    """Agent to support image capture from RTSP cameras.
+
+    This Agent captures images and writes them to a feed, as well as saving frames
+    to disk.  The camera can also be triggered to record....
+
+    Args:
+        agent (OCSAgent): OCSAgent object from :func:`ocs.ocs_agent.init_site_agent`.
+        directory (str): The path to image storage for this camera.
+        address (str): The hostname or IP address of the camera.
+        user (str): The user name for camera access.
+        password (str): The password for camera access.
+        seconds (int): The seconds between snapshots.
+        port (int): The RTSP port to use (default is standard 554).
+        urlpath (str): The remaining URL for the camera, which might include
+            options like channel and subtype.  This will depend on the manufacturer.
+        jpeg_quality (int): The JPEG quality for snapshot images (0-100).
+        max_snapshot_files (int): The maximum number of snapshots to keep.
+        record_fps (float): The frames per second for recorded video.
+        record_duration (int): The number of seconds for each recorded video.
+        max_record_files (int): The maximum number of recordings to keep.
+        fake (bool): If True, ignore camera settings and generate fake video
+            for testing.
+
+    Attributes:
+        agent (OCSAgent): OCSAgent object from :func:`ocs.ocs_agent.init_site_agent`.
+        log (txaio.tx.Logger): Logger object used to log events within the
+            Agent.
+        lock (TimeoutLock): TimeoutLock object used to prevent simultaneous
+            commands being sent to hardware.
+
+    """
+
+    def __init__(
+        self,
+        agent,
+        directory,
+        address,
+        user,
+        password,
+        seconds=10,
+        port=554,
+        urlpath=None,
+        jpeg_quality=20,
+        max_snapshot_files=10000,
+        record_fps=20.0,
+        record_duration=60,
+        max_record_files=100,
+        disable_motion=False,
```

Comment:
Can you make a similar argument that disables motion based on some time range during the day? I figure we wouldn't need to record motion during work hours when people on site are continuously entering/exiting containers

Reply:
That would be a nice feature that I considered for a future update, however I guess the usefulness of this agent is somewhat limited if we are continuously triggering off motion during the day.  I'll work on adding a time of day start / stop setting.

Reply:
This is now implemented as `--motion_start` and `--motion_stop` options.  The format is an ISO style string with UTC offset.  For example, to only perform motion detection between 7pm and 6am Chile time, one would pass:
```
--motion_start 19:00:00-04:00
--motion_stop 06:00:00-04:00
```